app/cp_nigeria/helpers.py

```diff
@@ -186,3 +194,357 @@ def create_report_content(self, project):
+        """
+        capex_df = pd.merge(
+            self.cost_assumptions[self.cost_assumptions["Category"] != "Revenue"],
+            self.system_params[["label", "value"]],
+            left_on="Target",
+            right_on="label",
+            suffixes=("_costs", "_outputs"),
+            how="left",
+        )
+
+        capex_df["Total costs [USD]"] = capex_df["USD/Unit"] * capex_df["value"]
+
+        # TODO double check here that transformer and mounting structure will stay 0 (currencly included in VAT in
+        #  excel but not here)
+        vat_costs = (
+            capex_df.groupby("Category")["Total costs [USD]"].sum()[["Logistics", "Labour and soft costs"]].sum()
+        )
+        vat_percentage = self.financial_params["tax"][0]
+        capex_df.loc[len(capex_df)] = {
+            "Description": "VAT",
+            "Category": "Taxes",
+            "Qty": 1,
+            "Total costs [USD]": vat_costs * vat_percentage,
+        }
+        capex_df["Total costs [NGN]"] = capex_df["Total costs [USD]"] * self.exchange_rate
+
+        # Add the power supply system capex from the sim results (accounts for custom cost parameters instead of static)
+        system_capex_rows = [
+            {
+                "Description": row["supply_source"].replace("_", " ").title(),
+                "Category": "Power Supply System",
+                "Total costs [NGN]": row["value"],
+            }
+            for index, row in self.system_params[self.system_params["category"] == "costs_capex"].iterrows()
+        ]
+
+        capex_df = pd.concat([capex_df, pd.DataFrame(system_capex_rows)], ignore_index=True)
+
+        return capex_df  # capex_df['Total costs [USD]'].sum() returns gross CAPEX
+
+    def revenue_over_lifetime(self):
+        """
+        This method returns a wide table calculating the revenue flows over project lifetime based on the cost
+        assumptions for revenue together with the number of consumers and total demand of the system.
+        """
+        revenue_df = pd.merge(
+            self.cost_assumptions[self.cost_assumptions["Category"] == "Revenue"],
+            self.system_params[["label", "value", "growth_rate"]],
+            left_on="Target",
+            right_on="label",
+            suffixes=("_costs", "_outputs"),
+            how="left",
+        )
+
+        revenue_lifetime = self.growth_over_lifetime_table(
+            revenue_df, "USD/Unit", "Growth rate", ["Description", "Target"]
+        )
+
+        revenue_flows = revenue_lifetime.mul(self.system_lifetime, level=1)
+        revenue_flows.loc["operating_revenues_total"] = revenue_flows.sum()
+
+        return revenue_flows
+
+    def om_costs_over_lifetime(self):
+        """
+        This method returns a wide table calculating the OM cost flows over project lifetime based on the OM costs of
+        the system together with the annual cost increase assumptions.
+        """
+        shs_costs_om = 7 * self.exchange_rate
```

Comment:
Is this magic number the number of days in the week? If yes minimally add a comment about this, pack it in a variable otherwise